include/envoy/server/listener_manager.h

```diff
@@ -212,6 +213,14 @@ class ListenerManager {
    */
   using FailureStates = std::vector<std::unique_ptr<envoy::admin::v3alpha::UpdateFailureState>>;
   virtual void endListenerUpdate(FailureStates&& failure_states) PURE;
+
+  // TODO(junr03): once ApiListeners support warming and draining, this function should return a
+  // weak_ptr to its caller. This would allow the caller to verify if the
+  // ApiListener is available to receive API calls on it.
+  /**
+   * @return ApiListener* the server's API Listener if it exists, nullptr if it does not.
+   */
+  virtual ApiListener* apiListener() PURE;
```

Comment:
nit: I would use a type aliased optional reference wrapper here.

Reply:
updated. Out of curiosity do you prefer these because the clarity that the caller does not own the object, or because optional "forces" you to think about nullopt, or something else?

Also there is no consistent naming for these type aliases. I went for the one with precedent that seemed closest to the smart pointer naming convention. I can homogenize naming for these aliases in a follow up PR.

Reply:
Yes because it's clear that it's not a raw pointer that can be deleted.